backend/src/main/java/de/propertyfinder/backend/api/octoparse/OctoparseApiService.java

```diff
@@ -30,28 +29,51 @@
 
     private final AccessTokenUtil accessTokenUtil;
     private final ApiConnectionUtil apiConnectionUtil;
+    private final String BASEURL = "https://dataapi.octoparse.com/";
 
+    @Autowired
     public OctoparseApiService(AccessTokenUtil accessTokenUtil, ApiConnectionUtil apiConnectionUtil) {
         this.accessTokenUtil = accessTokenUtil;
         this.apiConnectionUtil = apiConnectionUtil;
     }
 
     public List<OctoparseApiDto> getAllProperties() throws JsonProcessingException {
 
-        String tokenUrl = "https://dataapi.octoparse.com/token";
+        String tokenUrl = BASEURL + "token";
         String getToken = accessTokenUtil.getToken(userName, passWord, tokenUrl);
         String accessToken = getValueFromJson(getToken, "access_token");
         log.info("AccessToken: " + accessToken);
 
         String allDataFromOctoparseApi = getAllDataFromOctoparse(accessToken);
-        return mapToOctoparseApiDto(allDataFromOctoparseApi);
+        List<OctoparseApiDto> octoparseApiDtoList = mapToOctoparseApiDto(allDataFromOctoparseApi);
+        List<OctoparseApiDto> octoparseApiDtoListFilterd = filterOctparseApiData(octoparseApiDtoList);
+
+        return octoparseApiDtoListFilterd;
+    }
+
+    public List<OctoparseApiDto> filterOctparseApiData(List<OctoparseApiDto> octoparseApiDtoList) {
+        String removeSpacesAndTabs = "[/\\sg]";
+        String removeSpacesTabsAndUnit = "[/\\sgm²]";
+        return octoparseApiDtoList.stream()
+                .filter(string -> !string.getPurchasePrice().isEmpty())
+                .map(propertyObject -> {
+                    propertyObject.setPropertyTyp(propertyObject.getPropertyTyp().replaceAll(removeSpacesAndTabs,""));
```

Comment:
this is really hard to read :D It would be easier if you split it in multiple methods :-)

Reply:
or another format :D

Reply:
I would also move the mapper into a seperate method.

```java
private OctoparseApiDto sanitizePropertyObject(PropertyObject propertyObject) {
    propertyObject.setPropertyTyp(propertyObject.getPropertyTyp().replaceAll(removeSpacesAndTabs,""));
    propertyObject.setPurchasePrice(propertyObject.getPurchasePrice().replaceAll("[/\\sg€]",""));
    propertyObject.setSize(propertyObject.getSize().replaceAll(removeSpacesTabsAndUnit,""));
    propertyObject.setRoomCount(propertyObject.getRoomCount().replaceAll(removeSpacesAndTabs,""));
    propertyObject.setId(propertyObject.getId().replaceAll(removeSpacesAndTabs,"").replaceFirst("Objekt-Nr.:WHPO\\|Scout-ID:",""));
    propertyObject.setUsableArea(propertyObject.getUsableArea().replaceAll(removeSpacesTabsAndUnit,""));
    propertyObject.setLandArea(propertyObject.getLandArea().replaceAll(removeSpacesTabsAndUnit,""));
    return propertyObject;
}
```

And then use a method reference in the stream:
```java
return octoparseApiDtoList.stream()
                .filter(string -> !string.getPurchasePrice().isEmpty())
                .map(this::sanitizePropertyObject)
                .collect(Collectors.toList());
```